### ISSUE TITLE
two issues with ``meteor create --vue --release 3.1.1-beta.1`` 

### ISSUE THREAD

**Issue by mattsouth** [meteor/meteor#13530]
If I create the vue skeleton using the above command the wizard asks me to select the skeleton to use (which it didnt do in earlier versions) and it installs jorgenvatle:vite-bundler@3.0.0 which works in development with a warning to upgrade to meteor-vite@2 in npm but doesnt work with the --production flag.  dropping back to using jorgenvatle:vite-bundler@2.3.2 works in development without the warning and also works with the --production flag.

**Comment by jankapunkt** [comment]
@JorgenVatle might know what's the best action here?

**Comment by mattsouth** [comment]
Another option is to use jorgenvatle:vite with vite6 instead of jorgenvatle:vite-bundler but that's a bigger change to the vue skeleton, and maybe others too.

**Comment by leonardoventurini** [comment]
@mattsouth are you currently blocked by this, or the workaround works for you? We will be evaluating if we need to update the vue skeleton, thanks for letting us know.

**Comment by mattsouth** [comment]
Im not personally blocked by this at the moment.  I think ensuring each skeleton installs without error and then runs in development and production is a sensible quality check for a release but I appreciate this is difficult thing to automate.

**Comment by JorgenVatle** [comment]
I'd be up to update the skeleton apps depending on `jorgenvatle:vite-bundler` to upgrade to the latest stable release of the package.

There's also some structuring changes due that should make things a little more clear when building Meteor apps with Vite. Namely refactoring the placement and naming of the Vite and Meteor entry modules to be more in line with the current documentation for Meteor-Vite.

**Comment by nachocodoner** [comment]
@JorgenVatle: I updated to latest the meteor-vite and related dependencies on this PR, https://github.com/meteor/meteor/pull/13559

I tested locally and seems to work fine, but another pair of eyes would be good.

Edit: with `--production` mode I get a white screen with some logs in the console tab.

![Image](https://github.com/user-attachments/assets/e43a2d16-746c-4eee-961e-a772fe8608ab)

![Image](https://github.com/user-attachments/assets/d8985761-a6ef-43aa-bc21-9679b550a3c6)

**Comment by JorgenVatle** [comment]
@nachocodoner Appears to be an issue introduced in `meteor-vite@3.0.0`. Give it a go with `meteor-vite@3.2.1` 🤞

**Comment by mattsouth** [comment]
https://gitlab.com/mattsouth/meteor-vue-3.1.1

This repo takes the vue skeleton from meteor@3.1.1 and successfully configures jorgenvatle:vite, vite6 with meteor-vite@3.2.1.  It works in development and production.  I found I needed to add 
```
    stubValidation: {
      warnOnly: true
    }
```
to the vite meteor plugin config in order to surpress a problem with ObserveMultiplexer that's raised in https://github.com/JorgenVatle/meteor-vite/issues/272

**Comment by nachocodoner** [comment]
> Appears to be an issue introduced in meteor-vite@3.0.0. Give it a go with meteor-vite@3.2.1 🤞

Yes. In the PR I updated the skeleton to use that version, but no luck. `meteor run --production` doesn't work well on latest versions.

**Comment by nachocodoner** [comment]
![Image](https://github.com/user-attachments/assets/14bece16-25e8-43da-bd2e-0beb554d237e)

@mattsouth: Your repository works for me. I am trying to check on the differences, because it seems I am using the same exact dependencies and configuration, but not working on my end.  I don't see vite generating the proper files on `_vite-bundle` context. @JorgenVatle, any thoughts on what could cause this?

**Comment by mattsouth** [comment]
@nachocodoner I find its worth doing a ``meteor reset``, deleting the _vite-bundle folder and removing the references added in the meteor-entry.js files to get a good signal about what's working when moving between different versions.

edit: the _vite-bundle and references in the meteor-entry.js files will get recreated when you next run meteor.

**Comment by nachocodoner** [comment]
Thanks for your insights. They helped me dive deeper into `meteor-vite` and understand that the latest version needed some tweaks in the Vite config and project structure. [I’ve pushed all the changes](
https://github.com/meteor/meteor/pull/13559) to make these example apps work.

However, I noticed the warning you mentioned, which you stated that can be resolved by using `stubValidation.warnOnly` in the Vite config file. These warnings still appear on my end, and they’re the only issue left in the setup. Let me know if there's a way to fix them. It would be great to address this and improve the developer experience when using the Vue and Solid skeletons.

![Image](https://github.com/user-attachments/assets/73444c2b-3842-49fe-8752-be7b3ff3fe9d)

**Comment by nachocodoner** [comment]
We will include this fix for skeletons in the 3.1.2 release, soon likely in the beta later this week.

Hopefully a solution to address that warning is available soon.

**Comment by nachocodoner** [comment]
This fix is part of the Meteor 3.1.2 patch version.

Make sure to update to the latest version. If the global Meteor installation still points to an older version, try fully reinstalling the tool by running rm -rf ~/.meteor followed by npx meteor.

@mattsouth / @JorgenVatle: In case the warning displayed on the logs is ever fixed, please reopen this issue so that we can update the defaults for skeletons. Thanks.

**Comment by jankapunkt** [comment]
@nachocodoner so I quickly created a new vue project from skeleton with 3.1.2 and it looks good, got the warning from above but apart from that it seems to run fine. Will contrinue to test with tailwind and the likes to see if it works

**Comment by JorgenVatle** [comment]
Hey @nachocodoner. Sorry I completely lost track of this.

The warning is caused by the Meteor bundle analyzer seeing ObserveMultiplexer exported as a global in the final server bundle for Mongo. But its value is only defined when running Meteor in test environments. Since I took over maintenance of the project I added some quick and dirty validators as the bundle analyzer at the time would at times not generate valid package imports, which would lead to obscure and hard to debug issues in production. So it's primarily a safety precaution so that issues will be brought up when it happens.

Currently we don't have any good way of determining which exports defined in `package.js` are "test-only" as they are just defined as  an undefined variable regardless of the context Meteor is running in.

Seeing the state of the project now, we could probably set it to warn only by default as I don't believe there has been any notable issues where actual exports are invalidly parsed in newer versions. Perhaps also update the error message to make it clear that some warnings can safely be ignored unless explicitly used within the project's codebase.

For now however, there is an option to suppress warning messages altogether for certain packages.
```ts
// vite.config.ts

export default defineConfig({
  plugins: [
    meteor({
      stubValidation: {
        ignorePackages: ['mongo']
      }
    })
  ]
})
```

**Comment by JorgenVatle** [comment]
In regards to the `--production` flag. There is some room for improvement as Vite will only build for production once when Meteor first starts up. Changes to files created by Vite aren't watched, so you would have to manually restart Meteor for changes to take effect in this mode.

I've opened an issue to address this with a future update.
- https://github.com/JorgenVatle/meteor-vite/issues/287

**Comment by harry-73** [comment]
Hi,

```
export default defineConfig({
  plugins: [
    meteor({
      stubValidation: {
        ignorePackages: ['mongo']
      }
    })
  ]
})
```

To remove the warning, `mongo` does not work but `meteor/mongo` yes.

@nachocodoner could be good to update the skeleton in the next meteor release to avoid this warning for newcomer.

Thanks.

**Comment by nachocodoner** [comment]
Thank you for the suggestion. I opened the PR to ignore the error from `meteor/mongo`. 

https://github.com/meteor/meteor/pull/13614. We will include it on next release.

**Comment by jankapunkt** [comment]
Just want to say thanks to everyone involved here to make this work again so we can continue to promote vue on Meteor!
